src/main/webapp/dev/js/common/questionRubric.js

```diff
@@ -147,11 +178,21 @@ function addRubricRow(questionNum) {
         if (!$(`.rubricCol-${questionNum}-${cols}`).length) {
             continue;
         }
+
+        // Create rubric weight cell
+        const rubricWeightFragment =
+            `<div class="rubricWeights-${questionNum}">
+                <input type="number" class="form-control nonDestructive margin-top-10px" value="0"
+                        id="rubricWeight-${questionNum}-${newRowNumber - 1}-${cols}"
+                        name="rubricWeight-${newRowNumber - 1}-${cols}" step="0.01" required>
```

Comment:
Similar to the MCQ and MSQ questions (https://github.com/TEAMMATES/teammates/pull/8917#discussion_r202514669), not having `required` attribute here will likely cause the following problem.

In case if a user first checks the `Choices are weighted` checkbox and then adds new row or column, then the weight cell added by the `add row` or `add column` button will not be `required` as we make the existing cells required when `Choices are weighted` checkbox is checked and remove the `required` attribute when it is unchecked. 

However if we keep the `required` attribute here, then the above scenario will not occur, and if weights are not assigned in the question at all, then that should not create a problem either as the browser does not check for `required` if the field is hidden.